src/osd/PG.cc

```diff
@@ -2705,6 +2705,9 @@ void PG::publish_stats_to_osd()
     if ((state & (PG_STATE_ACTIVE|PG_STATE_PEERED)) &&
 	!(info.stats.state & (PG_STATE_ACTIVE|PG_STATE_PEERED)))
       info.stats.last_became_peered = now;
+    if (!(state & PG_STATE_CREATING) &&
+	(info.stats.state & PG_STATE_CREATING))
+      just_created = true;
```

Comment:
what if we send the message from right here?  and then also send one if we receive an MOSDPGCreate message on PGs that are already active?  then we don't have a time/retry at all.